apps/total-typescript/src/lib/exercises.ts

```diff
@@ -43,34 +43,70 @@ export const getExerciseMuxPlaybackId = async (exerciseSlug: string) => {
   return z.string().nullish().parse(exerciseVideo)
 }
 
-export const getExercise = async (slug: string): Promise<Exercise> => {
-  const exercise = await sanityClient.fetch(
+export const getExerciseMedia = async (exerciseSlug: string) => {
+  const exerciseMedia = await sanityClient.fetch(
     groq`*[_type == "exercise" && slug.current == $slug][0]{
+      "slug": slug.current,
+      body,
+      "stackblitz": resources[@._type == 'stackblitz'][0].openFile,
+      "muxPlaybackId": resources[@->._type == 'videoResource'][0]-> muxAsset.muxPlaybackId,
+      "transcript": resources[@->._type == 'videoResource'][0]-> castingwords.transcript,
+      "solution": resources[@._type == 'solution'][0]{
+        body,
+        "stackblitz": resources[@._type == 'stackblitz'][0].openFile,
+        "muxPlaybackId": resources[@->._type == 'videoResource'][0]-> muxAsset.muxPlaybackId,
+        "transcript": resources[@->._type == 'videoResource'][0]-> castingwords.transcript,
+        "slug": slug.current,
+      }
+    }`,
+    {slug: `${exerciseSlug}`},
+  )
+
+  return exerciseMedia
+}
+
+export const getExercise = async (
+  slug: string,
+  includeMedia: boolean = true,
+): Promise<Exercise> => {
+  const query = groq`*[_type == "exercise" && slug.current == $slug][0]{
       _id,
       _type,
       _updatedAt,
       title,
       description,
-      body,
       "slug": slug.current,
-      "stackblitz": resources[@._type == 'stackblitz'][0].openFile,
-      "muxPlaybackId": resources[@->._type == 'videoResource'][0]-> muxAsset.muxPlaybackId,
-      "transcript": resources[@->._type == 'videoResource'][0]-> castingwords.transcript,
+      ${
+        includeMedia
+          ? `      
+        body,
+        "stackblitz": resources[@._type == 'stackblitz'][0].openFile,
+        "muxPlaybackId": resources[@->._type == 'videoResource'][0]-> muxAsset.muxPlaybackId,
+        "transcript": resources[@->._type == 'videoResource'][0]-> castingwords.transcript,
+      `
+          : ''
+      }
```

Comment:
ended up protecting the route explicitly with middleware versus selectively loading media